File: sdks/nodejs/examples/send-and-list.ts
// Run with: SMSPIT_API_KEY=... npm run example
import { Client } from '../src/index.js';

const apiKey = process.env.SMSPIT_API_KEY;
if (!apiKey) throw new Error('Set SMSPIT_API_KEY');

const client = new Client({
  baseUrl: process.env.SMSPIT_BASE_URL ?? 'http://localhost:8080',
  apiKey,
});

const message = await client.send('+8801700000000', 'SMSPit', 'Your OTP is 123456');
console.log(`Captured ${message.id} (otp: ${message.otp})`);

const replay = await client.replay(message.id);
console.log(`Replayed as ${replay.id}`);

const page = await client.list({ limit: 5 });
console.log(`Inbox has ${page.total} message(s) total, showing ${page.limit}:`);
for (const inboxMessage of page.messages) {
  console.log(`  - ${inboxMessage.id}: ${inboxMessage.from} -> ${inboxMessage.to}: ${inboxMessage.message}`);
}

File: sdks/python/examples/send_and_list.py
"""Run with: SMSPIT_API_KEY=... python examples/send_and_list.py"""

import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))

from smspit import Client

api_key = os.environ.get("SMSPIT_API_KEY")
if not api_key:
    raise SystemExit("Set SMSPIT_API_KEY")

client = Client(
    base_url=os.environ.get("SMSPIT_BASE_URL", "http://localhost:8080"),
    api_key=api_key,
)

message = client.send(to="+8801700000000", from_="SMSPit", message="Your OTP is 123456")
print(f"Captured {message.id} (otp: {message.otp})")

replay = client.replay(message.id)
print(f"Replayed as {replay.id}")

page = client.list(limit=5)
print(f"Inbox has {page.total} message(s) total, showing {page.limit}:")
for inbox_message in page.messages:
    print(f"  - {inbox_message.id}: {inbox_message.from_} -> {inbox_message.to}: {inbox_message.message}")
